### DISCUSSION TITLE
How to log user out if app was uninstalled on iOS 14?

### DISCUSSION THREAD

**Discussion by VladRDV** [invertase/react-native-firebase#4361] [Q&A]
If I uninstall my app and then install it agan, auth state is still there, even if I delete app from library and reset my phone.
(the app came from Testflight btw)

I am considering using [this](https://www.reddit.com/r/reactnative/comments/epq8tc/auth_state_still_persisting_in_reactnativeios/femmfiq?utm_source=share&utm_medium=web2x&context=3) solution, but it would be nice to use something more reliable and clean.

**Comment by mikehardy** [comment]
If you have used Sign In With Apple, the app does not get to log the user out, the user has to go into their Apple User Account settings, in to the Sign In With Apple area and remove the app there. Otherwise they'll always be able to just log in again.

You did not enter any other details about your question so I won't spend time guessing further

**Reply by VladRDV** [reply]
@mikehardy thanks for reply :)

You're right, I shoud have been more specific

The problem comes from FB/Google/Email authentication

The problem is, that after in app registration(Google/FB button or via email (with verification)), user goes to onboarding, and in case user decides to go with another account or an error occurs, there is no way to leave that part of the app

Since client does not want to have log out button in onboarding, uninstalling the app should be the way, but because of auth state preservation in ios, that is not helping in this situation
